### PR TITLE
[FIX] crm: Fix issues on lead probability

### PR DESCRIPTION
**Pull request by tfr-odoo**
compute no team lead probability based on all lead
=================================================

Before this commit, lead with no team_id set had their automated
probability computed based on the subset of lead with no team_id.

Since all lead should end up we a team_id unless they do not match
any team criteria. the subset of lead without team is not a good
subset to estimate the probability of a new lead.

After this commit, a lead with no team_id set yet will have its
automated probability computed based on the entire lead set

do not erase 0% auto proba during merge
====================================

Issue
-----

When a lead with an automated probability is merged with other lead,
and this probability is 0, it will be considered as null value
and will be erased by the value of the next lead with a probability > 0

The final lead gets a probability != automated_probability and is
considered is_automated_probability = False and thus the automated
proability will be never set on the probability anymore

Expected behavior:
-----------------
If the probability is auto on the master keep the auto probability
If the probability is manual on the master keep that manual value
even if it's 0
in other words never take the probability from the merged records
and always keep the master one

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Review by dbeguin — COMMENTED** [review]
Looks ok ! 
We could also take the opportunity to add a specific test to ensure leads without teams are correctly computed based on all leads (and not only on "Team=False") and taken correctly into account in frequency table. Currently, there are no leads without teams in the tests.

**Comment by tde-banana-odoo** [comment]
Rebased on 14.3 as 14.2 is dead

**Comment by tde-banana-odoo** [comment]
@robodoo r+ rebase-ff

**Comment by tde-banana-odoo** [comment]
@robodoo retry
